src/main/java/com/cloudbees/jenkins/support/filter/DefaultContentFilter.java

```diff
@@ -0,0 +1,277 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.cloudbees.jenkins.support.filter;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.Functions;
+import hudson.XmlFile;
+import hudson.model.AbstractItem;
+import hudson.model.ManagementLink;
+import hudson.model.Saveable;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.randname.RandomNameGenerator;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.servlet.ServletException;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Filters contents based on names provided by all {@linkplain SensitiveNameProvider known sources}.
+ *
+ * @see SensitiveNameProvider
+ * @since TODO
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+public class DefaultContentFilter extends ManagementLink implements ContentFilter, Saveable {
+
+    public static DefaultContentFilter get() {
+        return all().get(DefaultContentFilter.class);
+    }
+
+    private static final Logger LOGGER = Logger.getLogger(DefaultContentFilter.class.getName());
+    private static final String ALT_SEPARATOR = " » ";
+
+    private volatile boolean enabled;
+    private final Map<String, Replacer> mappings = new ConcurrentHashMap<>();
+    private final Set<String> stopWords = ConcurrentHashMap.newKeySet();
+
+    public DefaultContentFilter() {
+        stopWords.addAll(Arrays.asList(
+                "jenkins", "node", "master", "computer",
+                "item", "label", "view", "all", "unknown",
+                "user", "anonymous", "authenticated"
+        ));
+        load();
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    @DataBoundSetter
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        save();
+    }
+
+    public Map<String, String> getMappings() {
+        Map<String, String> map = new HashMap<>(mappings.size());
+        for (Map.Entry<String, Replacer> entry : mappings.entrySet()) {
+            map.put(entry.getKey(), entry.getValue().replacement);
+        }
+        return map;
+    }
+
+    public Set<String> getStopWords() {
+        return stopWords;
```

Comment:
Is it need to add stopWords from outside? if not it would be an unmodifiable Set

Reply:
I changed it to be unmodifiable. I initially made this modifiable because it would allow a power user to change the value in the console, but now you can just edit the config file itself, so it's not really necessary.